gnovm/pkg/gnolang/machine.go

```diff
@@ -81,6 +82,9 @@ type Machine struct {
 	// it is executed. It is reset to zero after the defer functions in the current
 	// scope have finished executing.
 	DeferPanicScope uint
+
+	// Test Coverage
+	Coverage *CoverageData
```

Comment:
Does the VM usually keep track of this sort of data? cc @mvertes

Reply:
I’ve added a this field to update executed lines directly on the `Machine`, but it might be better to separate this functionality if possible.

Maybe I can fetch the executed lines separately using a getter, and then handle them within the coverage package. This would eliminate the dependency and overhead on the `Machine` as well.